xc7/utils/prjxray_create_edges.py

```diff
@@ -822,37 +815,34 @@ def main():
 
         edge_set = set()
 
-        for loc in progressbar.progressbar(grid.tile_locations()):
-            gridinfo = grid.gridinfo_at_loc(loc)
-            tile_name = grid.tilename_at_loc(loc)
+        c2 = conn.cursor()
+        for tile_pkey, tile_type_pkey, grid_x, grid_y in progressbar.progressbar(c.execute("SELECT pkey, tile_type_pkey, grid_x, grid_y FROM tile")):
+            loc = (grid_x, grid_y)
 
             # Not a synth node, check if in ROI.
-            if use_roi and not roi.tile_in_roi(loc):
+            if use_roi and not (roi_range[0] <= loc[0] and loc[0] <= roi_range[2] and roi_range[1] <= loc[1] and loc[1] <= roi_range[3]):
                 continue
 
+            # Process PIPs
+            for pip_pkey, pip_name, pip_src_wire_pkey, pip_dst_wire_pkey in c2.execute("SELECT pkey, name, src_wire_in_tile_pkey, dest_wire_in_tile_pkey FROM pip_in_tile WHERE tile_type_pkey = (?)", (tile_type_pkey, )):
 
-            tile_type = db.get_tile_type(gridinfo.tile_type)
-
-            for pip in tile_type.get_pips():
-                if pip.is_pseudo:
-                    continue
-
-                if not pip.is_directional:
-                    # TODO: Handle bidirectional pips?
-                    continue
+                # No pseudo pips and bidirectional pips should be present here
+                # as they were skipped during import to the SQLite database
+                # in prjxray_form_channels.py
```

Comment:
I recommend we add the pseudo and bidirectional column's to the database and filter them here still.

Reply:
I think this should be done in a separate PR.

Reply:
Probably worth doing that prior to merge this one then.  Should be a small PR.

Reply:
This comment still stands

Reply:
I'll look into it.

Reply:
@litghost I've created a PR with the change you suggested: https://github.com/SymbiFlow/symbiflow-arch-defs/pull/604 Once merged it will require integration with this one (I marked the place in code as FIXME).